src/pep8ext_naming.py

```diff
@@ -78,6 +79,12 @@ def _err(self, node, code, **kwargs):
     return lineno, col_offset + 1, '%s %s' % (code, code_str), self
 
 
+def _ignored(name, ignore=None):
+    if name and ignore:
```

Comment:
Do we need to check whether `name` is non-empty?
- `fnmatch('', 'foo') == False
- `fnmatch('a', '*') == True
- `fnmatch('', '*') == True

So it's only the last case that would vary, and I think I would assume that we'd prefer `True` to `False` there.

Or are you just attempting to avoid the overhead of checking the ignored symbols when `name` is empty? Do we often see empty names? I didn't think that happened much (if at all).

Reply:
On the next diff lines you can see that it was already verified that `name` isn't empty or `None`. I just refactorized this here, but frankly I don't have a strong opinion on this!

Reply:
My bad, only `ignore` used to be verified. Although IIRC I added `name` in the check because I happened to have a `None` going through.

Reply:
If you can isolate that `None` case, it might be useful to track that as its own minor issue. It's possible we don't have a higher-level condition that we should tweak so avoid getting that `None` value, and fixing that could save us from doing other unnecessary work.

Reply:
Here is a traceback for when I remove that test with any Python file containing a `def Test(): pass`:

```
Traceback (most recent call last):
  File "/usr/lib64/python3.7/multiprocessing/pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "/usr/lib64/python3.7/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "pep8-naming/.venv/lib/python3.7/site-packages/flake8/checker.py", line 666, in _run_checks
    return checker.run_checks()
  File "pep8-naming/.venv/lib/python3.7/site-packages/flake8/checker.py", line 598, in run_checks
    self.run_ast_checks()
  File "pep8-naming/.venv/lib/python3.7/site-packages/flake8/checker.py", line 502, in run_ast_checks
    for (line_number, offset, text, check) in runner:
  File "pep8-naming/src/pep8ext_naming.py", line 178, in visit_tree
    for error in self.visit_tree(child):
  File "pep8-naming/src/pep8ext_naming.py", line 174, in visit_tree
    for error in self.visit_node(node):
  File "pep8-naming/src/pep8ext_naming.py", line 195, in visit_node
    for error in visitor_method(node, parents, ignore_names):
  File "pep8-naming/src/pep8ext_naming.py", line 326, in visit_functiondef
    if _ignored(name, ignore):
  File "pep8-naming/src/pep8ext_naming.py", line 83, in _ignored
    return any(fnmatch(name, i) for i in ignore)
  File "pep8-naming/src/pep8ext_naming.py", line 83, in <genexpr>
    return any(fnmatch(name, i) for i in ignore)
  File "pep8-naming/.venv/lib/python3.7/fnmatch.py", line 34, in fnmatch
    name = os.path.normcase(name)
  File "pep8-naming/.venv/lib/python3.7/posixpath.py", line 54, in normcase
    s = os.fspath(s)
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

What we can see here is that `FunctionArgNamesCheck.visit_functiondef` sends a `name=None` to `_ignored` in its `for`, but I'm not sure why and I'm not familiar enough with the codebase nor AST stuff to fix this in a timely manner. Can you help?

Reply:
Ah, I think it's just a side-effect of the way we optimistically attempt to extract the argument name in that one place. I'd suggest writing it as (in the current code):

```
@@ -317,9 +316,9 @@ class FunctionArgNamesCheck(BaseASTCheck):
                 return node, arg

         for arg, name in arg_name(node.args.vararg), arg_name(node.args.kwarg):
-            if name in ignore:
-               continue
-            if name is not None and name.lower() != name:
+            if name is None or name in ignore:
+                continue
+            if name.lower() != name:
                 yield self.err(arg, 'N803', name=name)
                 return
```

In other words, let's do an explicit `name is not None` check in this case just before the "ignore" check.

Reply:
Also note that the `continue` statement in the current code isn't correctly indented to a multiple of four. Let's fix that while we're touching the neighboring code. :)

Reply:
Done, thanks!